src/py42/services/_auth.py

```diff
@@ -29,12 +29,15 @@ def _get_credentials(self):
 
 
 class V3Auth(C42RenewableAuth):
-    def __init__(self, auth_connection):
+    def __init__(self, auth_connection, totp=None):
         super(V3Auth, self).__init__()
         self._auth_connection = auth_connection
+        self._totp = totp if callable(totp) else lambda: totp
 
     def _get_credentials(self):
         uri = u"/c42api/v3/auth/jwt"
         params = {u"useBody": True}
-        response = self._auth_connection.get(uri, params=params)
+        currentToken = self._totp()
+        headers = {"totp-auth": str(currentToken)} if currentToken else None
```

Comment:
Should we do a retry here in case the current token expired after getting the current token and making the request to get the v3 token?

Reply:
Tokens are still accepted by the server for a period of time (not sure exactly how long but pretty considerable) even after they have "expired" in the authenticator app, so in order for us to hit this we'd need a very significant amount of time between the user inputting the code and py42 actually using it. That said, I'm not opposed to having a retry, just not sure of the best way to implement it.

Reply:
I'm not sure what the exact window of time is in our, but it's typical for TOTPs to be valid for some amount of time _after_ a new one is generated, (i.e., for some period of time, there will actually be _two_ codes that work) to handle this scenario. I'm fine with leaving this as-is and adding something to handle it if we actually experience problems.